build_tools/cmake/iree_cc_test.cmake

```diff
@@ -107,19 +107,50 @@ function(iree_cc_test)
   set_property(TARGET ${_NAME} PROPERTY DIRECT_DEPS ${_RULE_DEPS})
 
   string(REPLACE "::" "/" _PACKAGE_PATH ${_PACKAGE_NS})
-  set(_NAME_PATH "${_PACKAGE_PATH}:${_RULE_NAME}")
-  add_test(
-    NAME
-      ${_NAME_PATH}
-    COMMAND
-      # We run all our tests through a custom test runner to allow temp
-      # directory cleanup upon test completion.
-      "${CMAKE_SOURCE_DIR}/build_tools/cmake/run_test.${IREE_HOST_SCRIPT_EXT}"
-      "$<TARGET_FILE:${_NAME}>"
-    WORKING_DIRECTORY
-      "${CMAKE_BINARY_DIR}"
+
+  # Case for cross-compiling towards Android.
+  if(ANDROID)
+    set(_ANDROID_REL_DIR "${_PACKAGE_PATH}/${_RULE_NAME}")
+    set(_ANDROID_ABS_DIR "/data/local/tmp/${_ANDROID_REL_DIR}")
+
+    # Define a custom target for pushing and running the test on Android device.
+    set(_RUN_NAME "run_${_PACKAGE_NAME}_${_RULE_NAME}_on_android_device")
```

Comment:
Our test targets generally have a naming convention that matches bazel. That's why we've "_NAME_PATH" below. Also the name _NAME_PATH kind of sucks (my fault) especially when used in several places. Can you make this match too? I think the _on_android_device suffix is fine, although maybe not necessary, but then this should be:

```
set(_TEST_NAME "${_PACKAGE_PATH}:${_RULE_NAME}")
if(ANDROID)
  set(_TEST_NAME "{_TEST_NAME}_on_android_device") # or just drop _on_android_device
  # ...
else(ANDROID)
  # ...
endif(ANDROID)

list(APPEND _RULE_LABELS "${_PACKAGE_PATH}")
set_property(TEST ${_TEST_NAME} PROPERTY LABELS "${_RULE_LABELS}")
```

Reply:
Updated the test name. Having the `_on_android_device` makes it very clear that the tests are running on Android device so I think it's worth keeping.